src/fidesops/service/connectors/query_config.py

```diff
@@ -301,13 +345,53 @@ def transform_param(p: Any) -> str:
         return query_str
 
     def dry_run_query(self) -> Optional[str]:
+        """Returns a text representation of the query."""
         query_data = self.display_query_data()
         text_clause = self.generate_query(query_data, None)
         if text_clause is not None:
             return self.query_to_str(text_clause, query_data)
         return None
 
 
+class SnowflakeQueryConfig(SQLQueryConfig):
+    """Generates SQL in Snowflake's custom dialect."""
+
+    def format_fields_for_query(
+        self,
+        fields: List[str],
+    ) -> List[str]:
+        """Returns fields surrounded by quotation marks as required by Snowflake syntax."""
+        return [f'"{field}"' for field in fields]
+
+    def format_clause_for_query(
+        self,
+        field_name: str,
+        operator: str,
+    ) -> str:
+        """Returns field names in clauses surrounded by quotation marks as required by Snowflake syntax."""
+        return f'"{field_name}" {operator} (:{field_name})'
```

Comment:
Adding the `()` around `:{field_name}` is an interesting difference to how the more vanilla SQL connectors work. It could be that currently in our Postgres and MySQL connectors `IN` functions are accepted without parenthesis.

Reply:
I don't think this is a snowflake thing, but an idiosyncracy of the the way the dialect is realized in sqlalchemy. the other dialects we've used seem to add the '()' around tuple IN values automatically.